OpenRA.Mods.Common/Effects/Missile.cs

```diff
@@ -47,28 +58,59 @@ class MissileInfo : IProjectileInfo
 		[Desc("Probability of locking onto and following target.")]
 		public readonly int LockOnProbability = 100;
 
-		[Desc("In n/256 per tick.")]
-		public readonly int RateOfTurn = 5;
+		[Desc("Horizontal rate of turn.")]
+		public readonly int RateOfTurn = 5; // Rename to HorizontalRateOfTurn
 
-		[Desc("Explode when following the target longer than this many ticks.")]
+		[Desc("Vertical rate of turn.")]
+		public readonly int VerticalRateOfTurn = 5;
+
+		[Desc("Run out of fuel after being activated this many ticks. Zero for unlimited fuel.")]
 		public readonly int RangeLimit = 0;
 
-		[Desc("Trail animation.")]
-		public readonly string Trail = null;
+		[Desc("Explode when running out of fuel.")]
+		public readonly bool ExplodeWhenEmpty = true;
 
-		[Desc("Interval in ticks between each spawned Trail animation.")]
-		public readonly int TrailInterval = 2;
+		[Desc("Altitude above terrain below which to explode. Zero effectively deactivates airburst.")]
+		public readonly WDist AirburstAltitude = WDist.Zero;
+
+		[Desc("Cruise altitude. Zero means no cruise altitude used.")]
+		public readonly WDist CruiseAltitude = new WDist(512);
+
+		[Desc("Activate homing mechanism after this many ticks.")]
+		public readonly int ActivationDelay = 0;
```

Comment:
Just 'Activation' is a little too ambigous, in my opinion. Some modders might confuse this with `Arm`.
`HomingActivationDelay` would leave no room for misunderstandings.